### PR TITLE
Add ArrayValues type

### LINKED ISSUE #670: Proposal: `IndexesOf` and `ItemsOf`
This is a way to have typed indexes of a tuple, useful when you have types deriving of a `const` value

definition:
```ts
type ItemOf<T extends readonly any[]> = T[number]
type IndexesOf<T extends readonly any[]> = Exclude<Partial<T>['length'], T['length']>
```

use case:
```ts
const WEEK_DAYS = [
  'Monday', 'Tuesday', 'Wednesday',
  'Thursday', 'Friday', 'Saturday',
  'Sunday' ] as const

type WeekDayName = ItemsOf<typeof WEEK_DAYS>
type WeekDay = IndexesOf<typeof WEEK_DAYS>

const getDayName = (day: WeekDay): WeekDayName => WEEK_DAYS[day]
```

### PR DESCRIPTION
**Pull request by lucasteles**
Fixes #670

### CODE REVIEW

File: source/array-values.d.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+Use-case: This type is useful when working with constant arrays or tuples and you want to enforce type-safety with their values.
+
+@example
+```
+import type {ArrayValues} from 'type-fest';
+
+type values = ArrayValues<['a', 'b', 'c']>;
+//=> 'a' | 'b' | 'c'
+
+```
```
Comment (sindresorhus) [inline source/array-values.d.ts:15]: Use the example from ArrayIndices here too.

File: source/array-values.d.ts [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+@category Array
+*/
+export type ArrayValues<Element extends readonly unknown[]> = IterableElement<Element>;
```
Comment (sindresorhus) [inline source/array-values.d.ts:19]: Why not just

```suggestion
export type ArrayValues<T extends readonly unknown[]> = T[number];
```

**Review by sindresorhus — CHANGES_REQUESTED** [review]

**Comment by lucasteles** [comment]
@sindresorhus fixed

**Comment by sindresorhus** [comment]
Can you fix the merge conflict?

**Pull request merged**